addons/info/src/components/types/PrettyPropType.js

```diff
@@ -0,0 +1,73 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import ObjectType from './ObjectType';
+import OneOfType from './OneOfType';
+import ArrayOf from './ArrayOf';
+import ObjectOf from './ObjectOf';
+
+import { TypeInfo } from './proptypes';
+
+const PrettyPropType = ({ propType, depth }) => {
+  if (!propType) {
+    return <span>unknown</span>;
+  }
+
+  const { name } = propType || {};
+
+  if (name === 'shape') {
```

Comment:
I'm not really liking the usage of `if`-statements followed by a list of `switch`-cases.
Can you explain why this is?

Can you decrease the size of this function by making it 1 `switch`?